src/GovUk.Education.ExploreEducationStatistics.Admin/Security/AuthorizationHandlers/ViewReleaseTeamAccessAuthorizationHandler.cs

```diff
@@ -0,0 +1,46 @@
+#nullable enable
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Security;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using Microsoft.AspNetCore.Authorization;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Security.SecurityClaimTypes;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Security.AuthorizationHandlers
+{
+    public class ViewReleaseTeamAccessRequirement : IAuthorizationRequirement
+    {
+    }
+
+    public class ViewReleaseTeamAccessAuthorizationHandler :
+        AuthorizationHandler<ViewReleaseTeamAccessRequirement, Publication>
+    {
+        private readonly AuthorizationHandlerResourceRoleService _authorizationHandlerResourceRoleService;
+
+        public ViewReleaseTeamAccessAuthorizationHandler(
+            AuthorizationHandlerResourceRoleService authorizationHandlerResourceRoleService)
+        {
+            _authorizationHandlerResourceRoleService = authorizationHandlerResourceRoleService;
+        }
+
+        protected override async Task HandleRequirementAsync(
+            AuthorizationHandlerContext context,
+            ViewReleaseTeamAccessRequirement requirement,
+            Publication publication)
+        {
+            if (SecurityUtils.HasClaim(context.User, AccessAllPublications))
+            {
+                context.Succeed(requirement);
+                return;
+            }
+
+            if (await _authorizationHandlerResourceRoleService
+                    .HasRolesOnPublication(
+                        context.User.GetUserId(),
+                        publication.Id,
+                        PublicationRole.Owner, PublicationRole.Approver))
+            {
+                context.Succeed(requirement);
+            }
```

Comment:
I think we usually unit test this within an `AuthorizationHandlerTests` class but didn't spot any unit tests added here. Do you think it's worth adding any for this?

Reply:
Added, thanks!  Also renamed the handler to be in line with other "ViewSpecificPublication<Stuff>AuthorizationHandler" classes.